src/config.rs

```diff
@@ -270,15 +266,22 @@ impl Configs {
     }
 
     pub fn write(&self) -> Result<()> {
+        // Ensure directory exists
         create_dir_all(self.root_config_path.parent().unwrap())?;
-        let mut file = if let Ok(file) = File::options().write(true).open(&self.root_config_path) {
-            file
-        } else {
-            File::create(&self.root_config_path)?
-        };
-        let serialized_config = serde_json::to_vec_pretty(&self.root_config)?;
-        file.write_all(serialized_config.as_slice())?;
-        file.sync_all()?;
+
+        // Create temporary file to write initial data to. This is to ensure updates
+        // are atomic. After writingn the tmp file, we will rename it to the final destination,
+        // which is an atomic operation.
+        let mut tmp_file_path = temp_dir();
+        tmp_file_path.push(self.root_config_path.file_name().unwrap());
```

Comment:
I think we should avoid `.unwrap()`, you should be able to attach `.context()` to it and then `?` to return it as a `Result::Err()`.